hr_holidays_imposed_days/models/hr_imposed_holidays.py

```diff
@@ -0,0 +1,121 @@
+                vals = {
+                    'name': rec.name,
+                    'date_from': rec.date_from,
+                    'date_to': rec.date_to,
+                    'employee_id': employee.id,
+                    'type': 'remove',
+                    'holiday_status_id': rec.status_id.id,
+                }
+                leave = created.create(vals)
+                leave._onchange_date_from()
+                created |= leave
+            if rec.auto_confirm:
+                created.action_validate()
+
+    def _get_number_of_days(self, date_from, date_to):
+        """Returns a float equals to the timedelta between
+           two dates given as string."""
+
+        from_dt = fields.Datetime.from_string(date_from)
+        to_dt = fields.Datetime.from_string(date_to)
+        timedelta = to_dt - from_dt
+        diff_day = timedelta.days + float(timedelta.seconds) / 86400
+        return diff_day
+
+    @api.onchange('date_from', 'date_to')
+    def onchange_dates(self):
+        """
+        If there are no date set for date_to, automatically set one
+        8 hours later than the date_from.
+        Also update the number_of_days.
+        """
+        # Beacause when setting date_from, date_to is setted to actual day and
+        # will trigger an error.
+        if not self.id and self.date_from and \
+           self.date_to is False or self.date_from > self.date_to:
+            date_to_with_delta = datetime.datetime.strptime(
+                self.date_from,
+                tools.DEFAULT_SERVER_DATETIME_FORMAT) + datetime.timedelta(
+                hours=8)
+            self.date_to = str(date_to_with_delta)
+
+        # date_to has to be greater than date_from
+        if ((self.date_from and self.date_to) and
+                (self.date_from > self.date_to)):
+            raise exceptions.Warning(
+                _('The start date must be anterior to the end date.'))
+
+        # No date_to set so far: automatically compute one 8 hours later
+        if self.date_from and not self.date_to:
+            date_to_with_delta = datetime.datetime.strptime(
+                self.date_from,
+                tools.DEFAULT_SERVER_DATETIME_FORMAT) + datetime.timedelta(
+                hours=8)
+            self.date_to = str(date_to_with_delta)
+
+        # Compute and update the number of days
+        if ((self.date_to and self.date_from) and
+                (self.date_from <= self.date_to)):
+            diff_day = self._get_number_of_days(self.date_from, self.date_to)
+            self.number_of_days = self.compute_nb_days(diff_day)
+        else:
+            self.number_of_days = 0
+
+    def compute_nb_days(self, diff):
+        return round(math.floor(diff))+1
+
+    @api.one
+    @api.constrains('date_from', 'date_to')
+    def _check_dates(self):
+        if self.date_from >= self.date_to:
+            raise ValidationError(
+                "The start date must be anterior to the end date.")
```

Comment:
@gurneyalex But given the code above, I doubt that we could save an imposed_holidays with wrong dates.